docs/DeveloperGuide.adoc

```diff
@@ -14,6 +14,20 @@ ifdef::env-github,env-browser[:outfilesuffix: .adoc]
 
 By: `Team PIMP (W10-B1)`      Since: `Sep 2017`      Licence: `MIT`
 
+== Introduction
+
+=== Software Outline
+
+PIMP (Personal Information Manager Program) is a command line interface (CLI) address book that allows users to maintain their contacts. This program allows users to store various properties of their contacts, such as their home addresses, email addresses, phone numbers, or even custom tags. The program has four main components, namely the `User Interface (UI)`, `Logic`, `Model` and `Storage` components. Addressbook data is stored locally in an .xml file. 
+
+=== Purpose of Developer Guide
+
+This developer guide aims to give a brief overview to any developer who wishes to modify this addressbook to include more features, or to better suit their own needs. This developer guide will help any aspiring developer set up a suitable Java environment to develop this program in, as well as understand the different components of the addressbook and how they interact with each other. The functions available to the user will also be included, and various implementation details will be described. Various test cases have been made available to other developers, and a brief overview of the usage of these tests will also be included. 
```

Comment:
I don't it is a "brief overview" per se because there are actually a lot of details covered in the design and implementation parts. I think we generally should avoid saying our guide as brief because the guide itself should be quite detailed. We don't want to shoot ourselves in the foot right haha